src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -453,16 +453,21 @@ module TcRecdUnionAndEnumDeclarations =
     // Bind other elements of type definitions (constructors etc.)
     //------------------------------------------------------------------------- 
 
-    let CheckUnionCaseName (cenv: cenv) (id: Ident) =
+    let CheckUnionCaseName (cenv: cenv) (id: Ident) hasRQAAttribute =
         let g = cenv.g
         let name = id.idText
         if name = "Tags" then
             errorR(Error(FSComp.SR.tcUnionCaseNameConflictsWithGeneratedType(name, "Tags"), id.idRange))
 
         CheckNamespaceModuleOrTypeName g id
 
-        if not (String.isLeadingIdentifierCharacterUpperCase name) && name <> opNameCons && name <> opNameNil then
-            errorR(NotUpperCaseConstructor(id.idRange))
+        if g.langVersion.SupportsFeature(LanguageFeature.LowercaseDUWhenRequireQualifiedAccess) then
+
+            if not (String.isLeadingIdentifierCharacterUpperCase name) && not hasRQAAttribute && name <> opNameCons && name <> opNameNil then
```

Comment:
I wonder if, in case if we are using correct language version (which supports it), we want to have a some friendlier message, something mentioning that lowercase DU case can be used with RQAAttribute?

Reply:
You mean adding a warning  info mentioning explicitly  like 

`warning(LowercaseDUWhenRequireQualifiedAccess(....))`

Reply:
Yeah, something which will mention that lowercase DU cases are possible to use together with RQAA.

Reply:
Cool I will add that :)

Reply:
Seems like there are many ways to raise a warning. Is there a existing one that I can reuse . Or can you point to some example. Thanks :)

Reply:
Took me sometime to realize how to the warning strings are generated but I go it now :) . Updated following your suggestion.

Reply:
Let me know if the warning message is not clear enough .